<!--
 Copyright 2022 Shadow Robot Company Ltd.
 This program is free software: you can redistribute it and/or modify it
 under the terms of the GNU General Public License as published by the Free
 Software Foundation version 2 of the License.
 This program is distributed in the hope that it will be useful, but WITHOUT
 ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
 more details.
 You should have received a copy of the GNU General Public License along
 with this program. If not, see <http://www.gnu.org/licenses/>.
-->

<launch>
  <arg name="rosbag_path" default=""/>
  <arg name="tf_topic_name" default="/tf"/>
  <arg name="static_tfs" default="false" />
  <arg name="tf_static_topic_name" default="/tf_static"/>
  <arg name="remapped_tf_topic_name" default="tf_bag"/>
  <arg name="remapped_tf_static_topic_name" default="tf_static_bag"/>
  <arg name="tf_name_regexes" default="[]"/>
  <arg name="tf_static_name_regexes" default="$(arg tf_name_regexes)"/>
  <arg name="tcp_nodelay" default="true"/>
  <arg name="rosbag_extra_args" default=""/>

  <arg unless="$(arg static_tfs)" name="rosbag_topics" value="$(arg tf_topic_name)"/>
  <arg if="$(arg static_tfs)" name="rosbag_topics" value="$(arg tf_topic_name) $(arg tf_static_topic_name)"/>
  <arg unless="$(arg static_tfs)" name="rosbag_remappings" value="$(arg tf_topic_name):=$(arg remapped_tf_topic_name)"/>
  <arg if="$(arg static_tfs)" name="rosbag_remappings" value="$(arg tf_topic_name):=$(arg remapped_tf_topic_name) $(arg tf_static_topic_name):=$(arg remapped_tf_static_topic_name)"/>
  <arg name="rosbag_args" value="$(arg rosbag_path) -l --topics $(arg rosbag_topics) $(arg rosbag_remappings) $(arg rosbag_extra_args)"/>

  <node unless="$(eval arg('rosbag_path') == '')" pkg="rosbag" type="play" name="rosbag_mocap_tf" args="$(arg rosbag_args)" output="screen" />
  <node pkg="sr_utilities_common" type="real_time_tf_republisher.py" name="real_time_tf_republisher" output="screen">
    <param name="bag_tf_topic_name" value="$(arg remapped_tf_topic_name)" />
    <param name="bag_tf_static_topic_name" value="$(arg remapped_tf_static_topic_name)" />
    <param name="tcp_nodelay" value="$(arg tcp_nodelay)" />
    <param name="static_tfs" value="$(arg static_tfs)" />
    <rosparam param="tf_name_regexes" subst_value="True">$(arg tf_name_regexes)</rosparam>
    <rosparam param="tf_static_name_regexes" subst_value="True">$(arg tf_static_name_regexes)</rosparam>
  </node>
</launch>
